subgraph/src/receipt.ts

```diff
@@ -79,6 +85,25 @@ export function createReceipt(event: ethereum.Event): void {
     log.debug('Creating receipt for base currency lock {}', [lockAddress])
     receipt.payer = event.transaction.from.toHexString()
     receipt.amountTransferred = event.transaction.value
+    // We cannot trust `event.transaction.value` because the purchase function could in fact
+    // be happening inside of a larger transaction whose value is not the amount transfered,
+    // In that case, we need to look up the GNPChanged event
+    if (logs) {
```

Comment:
It seems that part is only needed if the tx value is null

Reply:
That's what I initially thought, but in fact the `transaction.value` could also be "wrong". Imagine if a safe tx buys 2 keys but starts with a 3rd tx in the buddle which funds the safe? Thent the value would be wrong

Reply:
Yes in the case of a safe or a multicall then the value can be calculated that way. Also for multiple purchases the value is the total of all keys purchased.

Reply:
Actually, that is a very good point. For receipts we actually have a single receipt per transaction, even if it ends up minting multiple keys (similar to when you buy multiple things at once you only get a single receipt at the store).
So my approach of looking at the GNP is not great.

I guess I can loop over all the GNP changed events for now.

We _really_ need an event to be triggered for receipt, like `PurchaseReceipt()` that would include the required details: `payer`, `contract`, `totalPaid`, `hash`... etc.